_posts/2019-08-15-release-1.9.0.md

```diff
@@ -0,0 +1,352 @@
+
+The time to recover a batch (DataSet, Table API and SQL) job from a task
+failure was significantly reduced. Until Flink 1.9, task failures in batch
+jobs were recovered by canceling all tasks and restarting the whole job, i.e,
+the job was started from scratch and all progress was voided. With this
+release, Flink persists intermediate results at network shuffle boundaries and
+use this data to limit the recovery to only those tasks that are affected by
+the failure. So-called failover regions of tasks which are connected via
+pipelined data exchanges define the boundaries among which tasks are affected
+by failures. More details are available in
+[FLIP-1](https://cwiki.apache.org/confluence/display/FLINK/FLIP-1+%3A+Fine+Grained+Recovery+from+Task+Failures).
+
+![alt_text]({{site.baseurl}}/img/blog/release-19-flip1.png "Fine-grained Batch
+Recovery")
+
+To use this new failover strategy, you need to do the following settings:
+
+ * Make sure you have the entry `jobmanager.execution.failover-strategy:
+   region` in your `flink-conf.yaml`.
+ * Set the `ExecutionMode` to `BATCH` in the `ExecutionConfig`.
+
+**Note:** The configuration of the 1.9 distribution has that entry by default,
+but when reusing a configuration file from previous setups, you have to add it
+manually.
+
+The "Region" failover strategy also improves the recovery of “embarrassingly
+parallel” streaming jobs, i.e., jobs without any shuffle like keyBy() or
+rebalance. When such a job is recovered, only the tasks of the affected
+pipeline (failover region) are restarted. For all other streaming jobs, the
+recovery behavior is the same as in prior Flink versions.
+
+
+### State Processor API (FLIP-43)
+
+Up to Flink 1.9, accessing the state of a job from the outside was limited to
+the (still) experimental [Queryable
+State](https://ci.apache.org/projects/flink/flink-docs-stable/dev/stream/state/queryable_state.html).
+This release introduces a new, powerful library to read, write and modify
+state snapshots using the batch DataSet API. In practice, this means:
+
+ * Flink job state can be bootstrapped by reading data from external systems,
+   such as external databases, and converting it into a savepoint.
+ * State in savepoints can be queried using any of Flink’s batch APIs
+   (DataSet, Table, SQL), for example to analyze relevant state patterns or
+   check for discrepancies in state that can support application auditing or
+   troubleshooting.
+ * The schema of state in savepoints can be migrated offline, compared to the
+   previous approach requiring online migration on schema access.
+ * Invalid data in savepoints can be identified and corrected.
+
+The new State Processor API covers all variations of snapshots: savepoints,
+full checkpoints and incremental checkpoints. More details are available in
+[FLIP-43](https://cwiki.apache.org/confluence/display/FLINK/FLIP-43%3A+State+Processor+API)
+
+
+### Stop-with-Savepoint (FLIP-34)
```

Comment:
For this section I'd suggest to explicitly declare:
1. The behavior of the stop command is changed, now every stop must be accompanied by a savepoint, which may slow down the job stop process although more semantically complete.
2. Although `-p savepoint-path` is marked as optional to the command, w/o this option the setting of `state.savepoints.dir` will be used as savepoint path, which has no default value. So it's better to explicitly remind user that either add `-p savepoint-path` into their existing stop job script (if any), or specify the `state.savepoints.dir` property in the job configuration.

What do you think? Thanks.

PS. More details please refer to the discussion in [PR#9030](https://github.com/apache/flink/pull/9030)

Reply:
What do you think @aljoscha @kl0u?

Reply:
Looks correct to me. It's also mentioned in the release notes mentioned here: https://issues.apache.org/jira/browse/FLINK-13123.

I'll make sure it is included in the 1.9.0 release notes page that is currently in the works (https://issues.apache.org/jira/browse/FLINK-13712).

Reply:
I think that putting such detailed information into the release notes makes more sense.